### PR TITLE
socket-proxyd: support Type=notify

### PR DESCRIPTION
**Pull request by gdamjan**
update examples in the man page too

### CODE REVIEW

File: src/socket-proxy/socket-proxyd.c [hunk 1]
```diff
@@ -12,6 +12,7 @@
 #include "sd-daemon.h"
 #include "sd-event.h"
 #include "sd-resolve.h"
+#include "daemon-util.h"
```
Comment (yuwata) [inline src/socket-proxy/socket-proxyd.c]: Please do not mix sd- prefixes headers and others, and sort headers.

Comment (gdamjan) [reply src/socket-proxy/socket-proxyd.c]: is it ok now?

Comment (yuwata) [reply src/socket-proxy/socket-proxyd.c]: Yes. Thank you.

**Comment by yuwata** [comment]
> ps. should I also add `DynamicUser=yes` to the examples in the man page, so it doesn't run as root?

I do not think so. Whether running with privileged or unprivileged user is irrelevant to socket-proxyd.

**Review by yuwata — APPROVED** [review]
LGTM.

**Comment by gdamjan** [comment]
> > ps. should I also add `DynamicUser=yes` to the examples in the man page, so it doesn't run as root?
> 
> I do not think so. Whether running with privileged or unprivileged user is irrelevant to socket-proxyd.

I meant as a suggestion to best practices, and a bit more ease-of-mind

**Comment by yuwata** [comment]
> > > ps. should I also add `DynamicUser=yes` to the examples in the man page, so it doesn't run as root?
> > 
> > 
> > I do not think so. Whether running with privileged or unprivileged user is irrelevant to socket-proxyd.
> 
> I meant as a suggestion to best practices, and a bit more ease-of-mind

I prefer that such suggestions are documented at other space, and simply refer it from the man page (and possibly also from other pages).
But that's in general hard, so as a first step, maybe better to add a reference to systemd.exec(5), and a short sentence that there are many sandboxing features.

**Comment by poettering** [comment]
hmm, patch looks ok. but i wonder about the rationale?

**Comment by gdamjan** [comment]
> hmm, patch looks ok. but i wonder about the rationale?

To be frank, my only rationale is that I'd like to see all examples provided by systemd to use best practices, so Type=notify and DynamicUser=yes. Educating users at each step.

**Pull request merged**